src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -0,0 +1,295 @@
+    ) {
+        $this->defaultOptions = array_merge([
+            'entity_manager' => null,
+            'expr' => null,
+            'mapping' => [],
+            'exclude' => [],
+            'strip_null' => false,
+            'id' => null,
+            'evict_cache' => false,
+            'auto_mapping' => true,
+            'attribute_only' => false,
+        ], $defaultOptions);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        if (0 === \count($this->registry->getManagerNames())) {
+            return false;
+        }
+
+        $options = $this->getOptions($argument);
+        if (null === $options['class']) {
+            return false;
+        }
+
+        if ($options['attribute_only'] && !$options['has_attribute']) {
+            return false;
+        }
+
+        // Doctrine Entity?
+        $em = $this->getManager($options['entity_manager'], $options['class']);
+        if (null === $em) {
+            return false;
+        }
+
+        return !$em->getMetadataFactory()->isTransient($options['class']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $options = $this->getOptions($argument);
+
+        $name = $argument->getName();
+        $class = $options['class'];
+
+        $errorMessage = null;
+        if (null !== $options['expr']) {
+            $object = $this->findViaExpression($class, $request, $options['expr'], $options);
+
+            if (null === $object) {
+                $errorMessage = sprintf('The expression "%s" returned null', $options['expr']);
+            }
+            // find by identifier?
+        } else {
+            $object = $this->find($class, $request, $options, $name);
+            if (false === $object) {
+                // find by criteria
+                $object = $this->findOneBy($class, $request, $options);
+                if (false === $object) {
+                    if (!$argument->isNullable()) {
+                        throw new \LogicException(sprintf('Unable to guess how to get a Doctrine instance from the request information for parameter "%s".', $name));
+                    }
+
+                    $object = null;
+                }
+            }
+        }
+
+        if (null === $object && !$argument->isNullable()) {
+            $message = sprintf('"%s" object not found by the "%s" Argument Resolver.', $class, self::class);
+            if ($errorMessage) {
+                $message .= ' '.$errorMessage;
+            }
+
+            throw new NotFoundHttpException($message);
+        }
+
+        return [$object];
+    }
+
+    private function getManager(?string $name, string $class): ?ObjectManager
+    {
+        if (null === $name) {
+            return $this->registry->getManagerForClass($class);
+        }
+
+        return $this->registry->getManager($name);
```

Comment:
be careful about the behavior for an unsupported name. In case of having the ORM and the ODM in the same project, it is likely that they won't have the same set of supported names, and we don't want to break the `supports` method by throwing an exception when the other resolver instance could handle it.

Reply:
good catch